test_runner/regress/test_ssl.py

```diff
@@ -151,3 +152,67 @@ def cert_reloaded():
     requests.get(addr, verify=str(env.ssl_ca_file)).raise_for_status()
     cur_cert = ssl.get_server_certificate(("localhost", port))
     assert cur_cert == sk_cert
+
+
+def test_server_and_cert_metrics(neon_env_builder: NeonEnvBuilder):
+    """
+    Test metrics exported from http/https server and tls cert reloader.
+    """
+    neon_env_builder.use_https_pageserver_api = True
+    neon_env_builder.pageserver_config_override = "ssl_cert_reload_period='100 ms'"
+    env = neon_env_builder.init_start()
+
+    env.pageserver.allowed_errors.append(".*Error reloading certificate.*")
+
+    ps_client = env.pageserver.http_client()
+    ps_metrics = ps_client.get_metrics()
+
+    # 1. Test connection started metric.
+    filter_https = {"scheme": "https"}
+    old_https_conn_count = ps_metrics.query_one(
+        "http_server_connection_started_total", filter_https
+    ).value
+
+    addr = f"https://localhost:{env.pageserver.service_port.https}/v1/status"
+    requests.get(addr, verify=str(env.ssl_ca_file)).raise_for_status()
+
+    ps_metrics = ps_client.get_metrics()
+    new_https_conn_count = ps_metrics.query_one(
+        "http_server_connection_started_total", filter_https
+    ).value
```

Comment:
FYI there is get_metrics_value helper: https://github.com/neondatabase/neon/commit/4f7704af245b80d2c2883b993d5c4920e53dbf70